src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -313,13 +315,18 @@ ReturnCode_t DomainParticipantImpl::enable()
     }
 
     guid_ = part->getGuid();
-    rtps_participant_ = part;
 
-    rtps_participant_->set_check_type_function(
-        [this](const std::string& type_name) -> bool
-        {
-            return find_type(type_name).get() != nullptr;
-        });
+    {
+        std::lock_guard<std::mutex> _(mtx_gs_);
+
+        rtps_participant_ = part;
```

Comment:
From this point on, we could use `part`  instead of `rtps_participant_` / `get_rtps_participant()`